server/metadata/collection.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"context"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/tigrisdata/tigris/errors"
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/server/transaction"
+	ulog "github.com/tigrisdata/tigris/util/log"
+)
+
+// CollectionSubspace is used to store metadata about Tigris collections.
+type CollectionSubspace struct {
+	metadataSubspace
+}
+
+// CollectionMetadata contains collection wide metadata.
+type CollectionMetadata struct {
+	OldestSchemaVersion int32
```

Comment:
`OldestSchemaVersion` may not be that intuitive, something like `NoUpdateNeededVersion` Or `StableCollectionVersion`